labs/11/practice.tex

```diff
@@ -0,0 +1,84 @@
+
+\begin{figure}[H]
+    \centering
+    \begin{subfigure}[b]{.45\linewidth}
+    \includegraphics[width=\linewidth]{labs/11/images/bubbles_rec.png}
+    \caption{Samples get pulled apart by reconstruction loss}\label{fig:bubbles_rec}
+    \end{subfigure}
+    \begin{subfigure}[b]{.45\linewidth}
+    \includegraphics[width=\linewidth]{labs/11/images/bubbles_kl.png}
+    \caption{Samples get pushed towards zero by KL-divergence loss}\label{fig:bubbles_kl}
+    \end{subfigure}
+    \caption{Effects of the different loss components}
+    \label{fig:bubbles}
+\end{figure}
+
+\section{Generative Adversarial Network}
+\subsection{Introduction}
+\begin{figure}
+    \centering
+    \includegraphics[width=0.7\textwidth]{"labs/11/images/VAE and GAN".png}
+    \caption{Architecture of GAN}
+    \label{fig:my_label}
+\end{figure}
+
+As we can see in the figure. The GAN model is similar to the VAE model.
+
+Consider a guy who makes fake money. The $\hat{x}$ produced by sampler+generator can be thought as the fake money he makes. Then he sends the money to someone(the discriminator) who wants to tell apart fake money from the genuine one ($x$). If the fake money is not 'good' enough, the discriminator will sends some feedback(sent through backpropagation) and the guy can improve his skills(gradient descent), until the discriminator can not tell apart the fake one from the genuine one. 
+
+From the sampler we come up with some noise. On top of the sampler we have the generator which tries to create authentic data, similar to the decoder to VAE, both of which are generative networks. From the generator we get our estimate $\hat{x}$on the manifold. The estimate, together with our real $x$, is sent to a discriminator which tries to tell apart $x$ from $\hat{x}$ that is difference between fake and actual values. We can optimize the generator depending on what fools and does not fool the discriminator
+
+For generator, We map the latent space to input space $\begin{array}{r}{G : \mathcal{Z} \rightarrow \mathbb{R}^{n}}, {z \mapsto \hat{x}}\end{array}$
+
+For discriminator, we go from the original input ($x$) or the crafted input ($\hat{x}$) to a learned loss. $\begin{aligned} D : & \mathbb{R}^{n} \rightarrow(0,1) & x \vee \hat{x} \mapsto \ell \end{aligned}$
+
+\subsection{Model illustration}
+
+\begin{figure}
+    \centering
+    \includegraphics[width=0.7\textwidth]{"labs/11/images/GAN illustration".png}
+    \caption{Visualization of encoding and decoding process}
+    \label{fig:my_label}
```

Comment:
You cannot use the same label multiple times. This creates errors.

Reply:
I have changed the label names. Thanks